### ISSUE TITLE
Release 0.31 Task dockerPush did not depend on any other task anymore and not able to push to Registry with simple config

### ISSUE THREAD

**Issue by mszalbach** [palantir/gradle-docker#511]
## What happened?

With the newest release 0.31 my build job did not publish any docker image because the task dockerPush was always up-to-date.
This is related to https://github.com/palantir/gradle-docker/pull/506/commits/578dcb87bcdb5d2314196f4f1ac84895e39cac52

Before this change Task:dockerPush depends on Task:tag and therefore build an image and tag it.

My configuration is.

```
docker {
    name "nexus.de/project/project:${project.version}"
    files tasks.distZip.outputs
    buildArgs(['PROJECT_VERSION': "${project.version}"])
}
```

Task:dockerTagsPush did not depend on any task before and after the change. So it seems you have to call

`gradlew dockerTag dockerTagsPush`

Buit because I am not using any tags in the config nothing is pushed. But before the image with the name was published to the Docker Registry.

## What did you want to happen?
Would be nice if dockerTagsPush could depend on dockerTag like dockerPush did before. Also the simple configuration stated in the Plugin Readme should be able to push something to a Docker Registry.  
If this is not possible because of compatibility issues for other projects, it would be nice to have the Readme having a hint how it is indented to build and publish using this plugin, and the minimal requiered config needed.

Thanks and Regards
Marcel

**Comment by mrexmelle** [comment]
The change also breaks my current CI/CD. Did you manage to resolve this (i.e. the substitute for the previous `gradle dockerPush`) @mszalbach ?

**Comment by mszalbach** [comment]
Cause of the Holidays I did not spend any time on this problem until now. So I pinned the versions to 0.29 where it works.

My current understandig is you need to call gradlew dockerTag dockerTagsPush (or add a gradle depends on dockerTag for the dockerTagsPush task in the gradle file). Also use a "tag" config in the docker block and not rely on the "name" config alone.

I hope to have a better look this or next week and decide how to fix this with the current release or if I switch to something else.

**Comment by mszalbach** [comment]
The new version https://github.com/palantir/gradle-docker/releases/tag/0.32.0 seems to fix all of my problems and it works without any changes.

Thanks a lot.
